src/client/connect.rs

```diff
@@ -28,15 +25,6 @@ pub struct Connect<A, B, C, E> {
     _pd: PhantomData<(A, B)>,
 }
 
-/// Executor that will spawn the background connection task.
-pub trait ConnectExecutor<T, B>:
```

Comment:
I would like to keep this trait as it simplifies, its pretty much a trait alias. We do this in tower-buffer as well. Now that we have the background type we can simplify the TypeExecutor bound which you wanted. I'm also fine with this trait being public.